File: Backend/AstroGame.Shared/Models/Players/PlayerSpecies.cs
using System;
using System.Collections.Generic;
using AstroGame.Shared.Enums;
using Newtonsoft.Json;

namespace AstroGame.Shared.Models.Players
{
    public class PlayerSpecies
    {
        public Guid Id { get; set; }
        public Guid PlayerId { get; set; }
        public Guid SpeciesId { get; set; }

        public string EmpireName { get; set; }

        /// <summary>
        /// The PlanetType this species prefers. 100 % habitability
        /// </summary>
        public PlanetType PreferredPlanetType { get; set; }

        public List<PlayerSpeciesPerk> Perks { get; set; }

        [JsonIgnore] public Player Player { get; set; }
        public Species Species { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/players/player_species.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/enums/planet_type.dart';
import 'package:astrogame_app/models/players/player_species_perk.dart';

import 'package:json_annotation/json_annotation.dart';

import 'species.dart';

part 'player_species.g.dart';

@GuidConverter()
@JsonSerializable()
class PlayerSpecies {
  late Guid id;
  late Guid playerId;
  late Guid speciesId;
  late String empireName;
  late PlanetType preferredPlanetType;

  List<PlayerSpeciesPerk> perks = [];

  late Species species;

  PlayerSpecies();

  factory PlayerSpecies.fromJson(Map<String, dynamic> json) =>
      _$PlayerSpeciesFromJson(json);
  Map<String, dynamic> toJson() => _$PlayerSpeciesToJson(this);
}
